graylog2-server/src/main/java/org/graylog2/rest/resources/system/indices/RetentionStrategyResource.java

```diff
@@ -0,0 +1,150 @@
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+@Api(value = "System/Indices/Retention", description = "Index retention strategy settings")
+@Path("/system/indices/retention")
+@Produces(MediaType.APPLICATION_JSON)
+@RequiresAuthentication
+public class RetentionStrategyResource extends RestResource {
+    private final Map<String, Provider<RetentionStrategy>> retentionStrategies;
+    private final ClusterConfigService clusterConfigService;
+
+    @Inject
+    public RetentionStrategyResource(Map<String, Provider<RetentionStrategy>> retentionStrategies,
+                                     ClusterConfigService clusterConfigService) {
+        this.retentionStrategies = requireNonNull(retentionStrategies);
+        this.clusterConfigService = requireNonNull(clusterConfigService);
+    }
+
+    @GET
+    @Path("config")
+    @Timed
+    @ApiOperation(value = "Configuration of the current retention strategy",
+            notes = "This resource returns the configuration of the currently used retention strategy.")
+    public RetentionStrategySummary config() {
+        final IndexManagementConfig indexManagementConfig = clusterConfigService.get(IndexManagementConfig.class);
+        if (indexManagementConfig == null) {
+            throw new InternalServerErrorException("Couldn't retrieve index management configuration");
+        }
+
+        final String strategyName = indexManagementConfig.retentionStrategy();
+        final Provider<RetentionStrategy> provider = retentionStrategies.get(strategyName);
+        if (provider == null) {
+            throw new InternalServerErrorException("Couldn't retrieve retention strategy provider");
+        }
+
+        final RetentionStrategy retentionStrategy = provider.get();
+        @SuppressWarnings("unchecked")
+        final Class<RetentionStrategyConfig> configClass = (Class<RetentionStrategyConfig>) retentionStrategy.configurationClass();
+        final RetentionStrategyConfig config = clusterConfigService.get(configClass);
+
+        return RetentionStrategySummary.create(strategyName, config);
+    }
+
+    @PUT
+    @Path("config")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Timed
+    @ApiOperation(value = "Configuration of the current retention strategy",
+            notes = "This resource stores the configuration of the currently used retention strategy.")
+    public void config(@Valid @NotNull RetentionStrategySummary retentionStrategySummary) {
```

Comment:
This also needs `@ApiParam`, no text box for the input in the API browser right now.